### ISSUE TITLE
Bug in multiline YAML parsing when followed by comment

### ISSUE THREAD

**Issue by egphill** [trilinos/Trilinos#2090]
YAML input
```yaml
  Boundary Conditions:
    Bottom:
      Dirichlet:
        Sideset: bottom
        Field:   ES_POTENTIAL
        Value: |
          double r_sq = xin*xin+yin*yin;
          double factor = 0.5*1.e8*1.60217662e-19/(2*3.14159265358979323846*8.854187817e-12);
          ES_POTENTIAL= factor*log(r_sq) +3*xin-3*yin;
  # end Boundary Conditions
```
gives the error
```code
14: Error at line: 4 unknown symbol: #
14: Function Body:
14: double r_sq = xin*xin+yin*yin; 
14: double factor = 0.5*1.e8*1.60217662e-19/(2*3.14159265358979323846*8.854187817e-12);
14: ES_POTENTIAL= factor*log(r_sq) +3*xin-3*yin;
14: # end Boundary Conditions
```

**Comment by ibaned** [comment]
Confirmed, working on a fix

**Comment by ibaned** [comment]
@egphill I've come up with a fix, its in pull request #2093
